code/modules/modular_armor/modular.dm

```diff
@@ -172,20 +173,32 @@
 	if(!istype(module))
 		return FALSE
 
-	if(ismob(loc) && (user.r_hand != src && user.l_hand != src))
-		if(!silent)
-			to_chat(user, span_warning("You need to remove the armor first."))
-		return FALSE
-
 	if(!do_after(user, equip_delay, TRUE, user, BUSY_ICON_GENERIC))
 		return FALSE
 
+///Temporarily unequips and then requips the armor during the attach process to keep armor datums/actions up to date if needed
+/obj/item/clothing/suit/modular/proc/do_attach(mob/living/user, obj/item/armor_module/module)
+	var/needs_requip = user.get_item_by_slot(SLOT_WEAR_SUIT) == src
+	if(needs_requip)
+		unequipped(user, SLOT_WEAR_SUIT)
```

Comment:
this drops your weapon?

Reply:
Why would it drop your weapon? It's a call on the armor itself. The armor stays in place, it just momentarily pretends it's unequipped in order to clear out actions and armor values, and then reapply it.

Reply:
Oh wait you mean your suit storage uhhhh that's a good question

Reply:
Okay it doesn't.